components/testmonitor/testmonitor.app.mjs

```diff
@@ -1,11 +1,171 @@
+        const { data: projects } = await this.getProjects({
+          page: page + 1,
+        });
+        return projects.map(({
+          id: value, name: label,
+        }) => ({
+          label,
+          value,
+        }));
+      },
+    },
+    query: {
+      type: "string",
+      label: "Query",
+      description: "The [search query](https://docs.testmonitor.com/#section/Requests/Searching).",
+    },
+    testResultId: {
+      type: "integer",
+      label: "Test Result ID",
+      description: "The test result identifier.",
+      async options({
+        projectId, page,
+      }) {
+        const { data: testResults } = await this.getTestResults({
+          page: page + 1,
+          project_id: projectId,
+        });
+        return testResults.map(({
+          id: value, code: label,
+        }) => ({
+          label,
+          value,
+        }));
+      },
+    },
+    with: {
+      type: "string[]",
+      label: "With",
+      description: "Include object relations. Refer to [Using relations](https://docs.testmonitor.com/#section/Requests/Relations).",
+    },
+  },
   methods: {
-    // this.$auth contains connected account data
-    authKeys() {
-      console.log(Object.keys(this.$auth));
+    _apiUrl() {
+      return `https://${this.$auth.domain}.testmonitor.com/api/v1`;
+    },
+    _getHeaders() {
+      return {
+        "Authorization": `Bearer ${this.$auth.api_token}`,
+      };
+    },
+    async _makeRequest({
+      $ = this, path, ...opts
+    }) {
+      const config = {
+        url: `${this._apiUrl()}/${path}`,
+        headers: this._getHeaders(),
+        ...opts,
+      };
+
+      return axios($, config);
+    },
+    getIssue({
+      $, issueId,
+    }) {
+      return this._makeRequest({
+        $,
+        path: `issues/${issueId}`,
+      });
+    },
+    getIssues(params) {
+      return this._makeRequest({
+        path: "issues",
+        params,
+      });
+    },
+    getProject({
+      $, projectId,
+    }) {
+      return this._makeRequest({
+        $,
+        path: `projects/${projectId}`,
+      });
+    },
+    getProjects(params) {
+      return this._makeRequest({
+        path: "projects",
+        params,
+      });
+    },
+    getTestResult({
+      $, testResultId,
+    }) {
+      return this._makeRequest({
+        $,
+        path: `test-results/${testResultId}`,
+      });
+    },
+    getTestResults(params) {
+      return this._makeRequest({
+        path: "test-results",
+        params,
+      });
+    },
+    async *paginate({
+      fn, params = {}, maxResults = null,
+    }) {
+      let count = 0;
+      let lastPage = false;
+
+      do {
+        const {
+          data,
+          meta: { last_page },
+        } = await fn(params);
+        for (const d of data) {
+          yield d;
+        }
+
+        if (maxResults && ++count === maxResults) {
+          return count;
+        }
+        lastPage = !last_page;
+      } while (lastPage);
```

Comment:
I don't see that this is actually paginating. We need to use `page` and increment it to get each subsequent page. https://docs.testmonitor.com/#section/Requests/Pagination. Also, the `lastPage` prop isn't boolean. It's actually the page number of the last page.

```suggestion
    async *paginate({
      fn, params = {}, maxResults = null,
    }) {
      let count = 0;
      let lastPage = false;

      params = {
        ...params,
        page: 1,
      }

      do {
        const {
          data,
          meta: { last_page },
        } = await fn(params);
        for (const d of data) {
          yield d;
        }

        if (maxResults && ++count === maxResults) {
          return count;
        }

        lastPage = last_page == params.page;
        params.page++;
      } while (!lastPage);
    },
```